### ISSUE TITLE
Fun hack:  Support for callouts between two text shapes

### ISSUE THREAD

**Issue by kjambunathan** [kjambunathan/org-mode-ox-odt#173]
Cc @QiangF

Now I understand what is happening.  It has taken almost 5 years, and let me whether you understood it correctly.

You have been asking for subfigure and stuff NOT in the context for preparing academic / formal articles, but more for common folks.  Long story short, **I am willing to support for arrows or callouts between two shapes** .    This is mostly for  fun hack and play. 

In this comment https://github.com/kjambunathan/org-mode-ox-odt/issues/1#issuecomment-315737241, I remember seeing some "complex" shapes and stuff and at that point in time it seemed very complex for what the ODT exporter could do, or what my limited understanding of  what your requirement was.    **(Unfortunately, since you uploaded to `tinyupload` those files are not accessible at all, and I am not able to recall clearly what the documents were. If you still have that document, or can create it by hand, please send it to me)**

The ["monkey with banana remark"](https://github.com/kjambunathan/org-mode-ox-odt/issues/109#issuecomment-946294865) below,  gives me an impression that the target audience for your documents are not some professor or a grown adult,  but more illustrations in popular magazines or even instructional material for little children.  **I can see the usefulness of requirement and I am willing to support "arrows or callouts between two shapes"**

> No. It's just another stupid requirement of some chinese magzines. Their rationale is that sometimes subpictures have a relationship, for examples an image has an arrow pointing down to another sub image, (a zoo keeper is holding a banana, and the monkey is in the figure below), the arrow is not supposed to point to subcaptions, so it looks better to separate sub-picture with their descriptions. The description is labeled in the main caption. I believe insisting on this kind of trivial thing is wrong, but powerful people can do whatever they want.

**Comment by QiangF** [comment]
You are correct. :-)  There are professional document that put text annotation over figures too, for example, https://tex.stackexchange.com/questions/454660/adding-a-label-on-top-of-figure

**Comment by kjambunathan** [comment]
> https://github.com/kjambunathan/org-mode-ox-odt/commit/a287a42fc777857394a46698b86c36e1a0fb0762

Unit test case for this feature: [draw-custom-shapes-and-connect-them.zip](https://github.com/kjambunathan/org-mode-ox-odt/files/8310727/draw-custom-shapes-and-connect-them.zip)

To get this

![Screenshot from 2022-03-20 12-00-38](https://user-images.githubusercontent.com/19869864/159151205-c24b3169-b591-44e0-b1ef-379550e09038.png)

do this

```
#+options: toc:nil

#+odt_validate: abort

* A Rectangle without any embedded image

#+ATTR_ODT: :width 5 :height 5
#+begin_customshape
    Ipsum excepteur sint pariatur irure esse
#+end_customshape

* A Rectangle with an embedded image

#+ATTR_ODT: :width 5 :height 5 :image "images/org-mode-unicorn.png"
#+begin_customshape
    Enim sint esse qui sed
#+end_customshape

* An Ellipse with an embedded image

#+ATTR_ODT: :width 5 :height 5 :image "images/org-mode-unicorn.png" :shape "ellipse"
#+begin_customshape
    Enim sint esse qui sed
#+end_customshape

* Two shapes connected with an arrow

|        | [fn:1] |
| [fn:2] |        |

[fn:1]

#+ATTR_ODT: :width 5 :height 5 :image "images/org-mode-unicorn.png" :other-id ":8<->bottomleft:1"
#+NAME: topright
#+begin_customshape
    Top Right
#+end_customshape

[fn:2]

#+ATTR_ODT: :width 5 :height 5 :image "images/org-mode-unicorn.png"  :shape "ellipse"
#+NAME: bottomleft
#+begin_customshape
    Bottom Left
#+end_customshape

* COMMENT Text overflows the shape that has no embedded image

#+ATTR_ODT: :width 5 :height 5
#+begin_customshape
Duis est veniam, laboris deserunt ex elit, minim sit irure et.
Cupidatat velit eu occaecat consequat adipiscing mollit nisi
exercitation ut consectetur consequat labore ipsum in lorem sunt dolor
eiusmod ut.  Lorem occaecat labore labore dolor officia ex commodo do
nisi laborum magna proident, lorem excepteur commodo excepteur officia
amet,.  Mollit sed elit, labore commodo ipsum laboris sunt velit
reprehenderit aute cupidatat sed deserunt et cupidatat culpa voluptate
adipiscing exercitation labore.
#+end_customshape

* COMMENT Text overflows the shape that has no embedded image

#+ATTR_ODT: :width 5 :height 5 :image "images/org-mode-unicorn.png"
#+begin_customshape
Esse eiusmod enim veniam, in proident, mollit mollit quis esse culpa
velit velit tempor dolor consequat nulla sed.  Officia id consectetur
eu adipiscing in cupidatat sed occaecat dolore deserunt enim dolor
fugiat reprehenderit.  Est aute minim velit incididunt anim cillum
excepteur sint et qui elit, pariatur culpa magna non eiusmod
cupidatat.  Est nulla ipsum esse adipiscing anim officia velit ullamco
voluptate officia in incididunt magna exercitation mollit veniam,
incididunt elit, eu consectetur irure adipiscing.  Voluptate aute et
velit elit, quis velit voluptate.
#+end_customshape
```

**Comment by kjambunathan** [comment]
> No. It's just another stupid requirement of some chinese magzines. Their rationale is that sometimes subpictures have a relationship, for examples an image has an arrow pointing down to another sub image, (a zoo keeper is holding a banana, and the monkey is in the figure below), the arrow is not supposed to point to subcaptions, so it looks better to separate sub-picture with their descriptions. The description is labeled in the main caption. I believe insisting on this kind of trivial thing is wrong, but powerful people can do whatever they want.

I have added support for adding drawing arrows between the shapes.   Note that I have put them in a table so that you can place them "relative" to each other without having to specify the `svg:x` and `svg:y` values.    The shapes are movable, and they would still remain connected if you move them.  In order that you move the shapes, change their anchors from `as-char` to something else say  `paragraph`.  This way you can dislodge them from their tables, and move them freely around.

**(This was a fun hack.  I never imagined that I would end up doing something like this.)**

**Comment by kjambunathan** [comment]
>  There are professional document that put text annotation over figures too

I just looked at my browser history and what really convinced me was this line 

> **Note that often IAENG papers with subfigures do not employ subfigure captions (using the optional argument to \subfloat), but instead will reference/describe all of them (a), (b), etc., within the main caption**

from this article https://www.cnblogs.com/code-saturne/p/10565151.html#sec-16-6

**Comment by QiangF** [comment]
This is great for illustration purposes. I will definitely make more good looking slides with this feature.
There are other ways of create flow chart and simple shapes using plain text in an org mode file, such as ditta, which used to come with org distribution.
Many other options exist, like aafigure, svgbob, markdeep. They are very handy in creating perfectly aligned and good look illustrations when combined with Emacs artist-mode.
For more complicated design, I think overlay those vector images on top of a background figure is much more simple than working with libreoffice xml files, or libreoffice draw. The benifit of libreoffice over external programs (also imagemagic) is the consistency in font and font size.
https://github.com/aafigure/aafigure
https://github.com/ivanceras/svgbob
https://casual-effects.com/markdeep

**Comment by kjambunathan** [comment]
> https://github.com/aafigure/aafigure
> https://github.com/ivanceras/svgbob
> https://casual-effects.com/markdeep

There is also [ Libo_PlantUML:   This extension enables PlantUML directly in your Writer documents.](https://extensions.libreoffice.org/en/extensions/show/libo_plantuml).  

Very very long time back I have tried it.
